resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh

```diff
@@ -96,22 +96,6 @@ case "$SPARK_K8S_CMD" in
       "$@"
     )
     ;;
-  driver-py)
```

Comment:
I think this is fine and what we want to do. But at some point we're going to want to make the API between spark submit and launched containers stable.

Using this as an example, if a user upgraded their spark-submit version to 3.0 but didn't upgrade the version of Spark in their docker image, the docker container's command will attempt to look up these old environment variables that are no longer being set by spark submit.

We should be thinking about making this contract stable from 3.0 onwards. For this coming release I think this is fine.

Reply:
> if a user upgraded their spark-submit version to 3.0 but didn't upgrade the version of Spark in their docker image

I don't think Spark really supports that in any case. It is expected that you're using the same version of Spark in the spark-submit side and on the other side, regardless of the cluster manager. e.g., imagine client mode in the scenario you're talking about.

But definitely, the contract here needs to be well defined (I've asked for this a long time ago and that's one of the reasons why k8s is still experimental).